extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/graal/JdkSubstitutions.java

```diff
@@ -8,24 +8,42 @@
 import com.oracle.svm.core.annotate.Alias;
 import com.oracle.svm.core.annotate.Substitute;
 import com.oracle.svm.core.annotate.TargetClass;
+import com.oracle.svm.core.jdk.JDK11OrLater;
+import com.oracle.svm.core.jdk.JDK8OrEarlier;
 
-@TargetClass(className = "sun.misc.URLClassPath$Loader")
+@TargetClass(className = "sun.misc.URLClassPath$Loader", onlyWith = JDK8OrEarlier.class)
 final class Target_sun_misc_URLClassPath$Loader {
 
     @Alias
     public Target_sun_misc_URLClassPath$Loader(URL url) {
     }
 }
 
-@TargetClass(className = "sun.misc.URLClassPath$FileLoader")
+@TargetClass(className = "jdk.internal.loader.URLClassPath$Loader", onlyWith = JDK11OrLater.class)
+final class Target_jdk_internal_loader_URLClassPath$Loader {
+
+    @Alias
+    public Target_jdk_internal_loader_URLClassPath$Loader(URL url) {
+    }
+}
+
+@TargetClass(className = "sun.misc.URLClassPath$FileLoader", onlyWith = JDK8OrEarlier.class)
 final class Target_sun_misc_URLClassPath$FileLoader {
 
     @Alias
     public Target_sun_misc_URLClassPath$FileLoader(URL url) throws IOException {
     }
 }
 
-@TargetClass(className = "sun.misc.URLClassPath")
+@TargetClass(className = "jdk.internal.loader.URLClassPath$FileLoader")
```

Comment:
This class doesn't exist in JDK8, is that missing a `onlyWith=JDK11OrLater.class`?